src/main/java/org/variantsync/diffdetective/variationtree/VariationNode.java

```diff
@@ -0,0 +1,583 @@
+        }
+        return getParent().getIfNode();
+    }
+
+    /**
+     * Returns {@code true} iff this node is the parent of the given node.
+     *
+     * @see getChildren
+     */
+    public boolean isChild(VariationNode<T> child) {
+        return child.getParent().isSameAs(child);
+    }
+
+    /**
+     * Returns the index of the given child in the list of children of this node.
+     * Returns -1 if the given node is not a child of this node.
+     *
+     * <p>Warning: If this is a {@link DiffNode.Projection}, then the returned index may be
+     * different to the index returned by {@link DiffNode#indexOfChild}.
+     *
+     * @see getChildren
+     */
+    public int indexOfChild(final VariationNode<T> child) {
+        return getChildren().indexOf(child);
+    }
+
+    /**
+     * The same as {@link addChild(VariationNode<T>, int)} but puts the node at the end of the
+     * children list instead of inserting it at a specific index.
+     *
+     * @see getChildren
+     */
+    public abstract void addChild(final VariationNode<T> child);
+
+    /**
+     * Adds a child before the given index to the children list of this node and sets its parent to
+     * this node.
+     *
+     * <p>When calling {@link indexOfChild} with {@code child} the returned index will be
+     * {@code index} as long as the children list isn't modified.
+     *
+     * @see addChildren
+     */
+    public abstract void addChild(final VariationNode<T> child, int index);
+
+    /**
+     * Adds the given nodes in traversal order as children using
+     * {@link addChild(VariationNode<T>)}.
+     */
+    public void addChildren(final Collection<VariationNode<T>> children) {
+        for (final var child : children) {
+            addChild(child);
+        }
+    }
+
+    /**
+     * Removes the given node from this node's children list and sets the parent of {@code child}
+     * to {@code null}.
+     *
+     * @return {@code true} iff the child was removed, {@code false} iff {@code child} is not a
+     * child of this node
+     * @see removeChildren
+     * @see getChildren
+     */
+    public abstract boolean removeChild(final VariationNode<T> child);
+
+    /**
+     * Removes the given nodes from the children list using {@link removeChild}.
+
+     * @see removeAllChildren
+     */
+    public void removeChildren(final Collection<VariationNode<T>> childrenToRemove) {
+        for (final var childToRemove : childrenToRemove) {
+            removeChild(childToRemove);
+        }
+    }
+
+    /**
+     * Removes all children of this node and sets their parent to {@code null}.
+     * Afterwards, this node will have no children.
+     */
+    public abstract void removeAllChildren();
+
+    /**
+     * Adds this subtree below the given parent.
+     * Inverse of {@link drop}.
+     *
+     * @see addChild
+     */
+    public void addBelow(final VariationNode<T> parent) {
+        if (parent != null) {
+            parent.addChild(this);
```

Comment:
We also have to check that `this` does not have have a parent already.

Reply:
This is already checked by `addChild`. Documentation for this behavior is in progress.

Reply:
Functions which check their arguments are now documented with `throws IllegalArgumentException`.

Reply:
Good idea!